pandas/tests/dtypes/cast/test_infer_dtype.py

```diff
@@ -188,11 +189,32 @@ def test_infer_dtype_from_array(arr, expected, pandas_dtype):
         (Period("2011-01-01", freq="D"), object),
     ],
 )
-def test_cast_scalar_to_array(obj, dtype):
-    shape = (3, 2)
-
+@pytest.mark.parametrize("shape", [(), (5,), (3, 2)])
+def test_cast_scalar_to_array(obj, dtype, shape):
     exp = np.empty(shape, dtype=dtype)
     exp.fill(obj)
 
-    arr = cast_scalar_to_array(shape, obj, dtype=dtype)
+    arr = cast_scalar_to_array(shape, obj, dtype=np.dtype(dtype))
     tm.assert_numpy_array_equal(arr, exp)
+
+
+@pytest.mark.parametrize(
+    "obj_in,dtype_in,obj_out,dtype_out",
+    [
+        (NaT, "timedelta64[ns]", np.timedelta64("NaT"), "timedelta64[ns]"),
+        (Timedelta(1), "timedelta64[ns]", 1, "timedelta64[ns]"),
+        (NaT, "datetime64[ns]", np.datetime64("NaT"), "datetime64[ns]"),
+        (Timestamp(1), "datetime64[ns]", 1, "datetime64[ns]"),
+        (Timestamp(1, tz="US/Eastern"), "datetime64[ns]", 1, "datetime64[ns]"),
```

Comment:
this is a deeply weird behavior that we should be trying to deprecate

Reply:
Ok, true, it can be counterintuitive that `pd.Timestamp("1970-01-01", tz="US/Pacific") != pd.Timestamp(0, tz="US/Pacific")`